### PR TITLE
Faster docker-compose sessions by optimizing the settings script

### PR DESCRIPTION
**Pull request by RickHawesUSDS**
This PR improves developer ergonomics a little by speeding up the `docker-compose` session startup time by 12-15 seconds. 

## Changes
- Added a HEAD method on the `/settings/organizations` resource that returned the max(createdAt) value in a Last-Modified header
- The settings script compares the `organization-local.yml` modified date from the returned modified date and exits early if no updates are needed. 
- Updated openapi.yml

## Alternatives Considered
One could start setting up profiles in the `docker-compose.yml` script to only run the settings service if a specific profile was set. This would lead to further session startup time improvements, but it would also require developers to learn how to use the profiles. Also, many profiles should be considered (backend, test, frontend, all, etc...). 

### Testing
- [x] Tested locally?
- [x] Ran `quickTest all`?
- [x] Ran `./prime test` against local Docker ReportStream container?
- [ ] Downloaded a file from `http://localhost:7071/api/download`?
- [ ] Added tests?

### Security
- [ ] Did you check for sensitive data, and remove any? 
- [ ] Does logging contain sensitive data?  
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

### Process
- [ ] Includes a summary of what a code reviewer should verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Security
N/A

## Fixes
No issue. Just an optimization that occurred to me. 

## To Be Done
Lot's more optimizations can be done with the full support of `Last-Modified` and `If-Modified-Since` headers. Yahoo did this as a common practice on their APIs.

### CODE REVIEW

File: prime-router/docker-compose.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ services:
 
   settings:
     build: settings/.
-    command: "--wait=8 prime_dev /settings/organizations-local.yml"
+    command: "--wait=8 --check-last-modified prime_dev /settings/organizations-local.yml"
     depends_on:
```
Comment (RickHawesUSDS) [inline prime-router/docker-compose.yml:76]: new option in the settings script.

File: prime-router/docs/openapi.yml [hunk 2]
```diff
@@ -100,12 +100,33 @@ paths:
...
         '200':
           description: OK
+          headers:
+            Last-Modified:
+              description: The date and time any setting was modified in GMT.
+              example: Wed, 21 Oct 2015 07:28:00 GMT
+              required: false
+              schema:
+                type: string
           content:
```
Comment (RickHawesUSDS) [inline prime-router/docs/openapi.yml:110]: The `Last-Modified` header is an HTTP standard header.

File: prime-router/settings/put-local-settings.py [hunk 3]
```diff
@@ -51,14 +55,31 @@ def put_receiver(receiver):
...
     r.raise_for_status()
-    
+
+
+def get_last_modified():
+    url = f'{base_url}/settings/organizations'
+    r = requests.head(url, headers=headers)
+    r.raise_for_status()
+    if 'Last-Modified' in r.headers:
+        return parsedate_to_datetime(r.headers['Last-Modified'])
```
Comment (RickHawesUSDS) [inline prime-router/settings/put-local-settings.py:65]: Email and Last-Modified use the same date format.

File: prime-router/settings/put-local-settings.py [hunk 4]
```diff
@@ -51,14 +55,31 @@ def put_receiver(receiver):
...
     try:
         if args.wait > 0:
             print(f"Waiting for {args.wait} seconds")
             time.sleep(args.wait)
 
+        if args.check_last_modified:
+            settings_modified = get_last_modified()
+            file_modified = os.path.getmtime(org_file)
+            if settings_modified.timestamp() >= file_modified:
+                print(f"Settings modified {settings_modified} after input file, do not update settings")
```
Comment (RickHawesUSDS) [inline prime-router/settings/put-local-settings.py:80]: Pretty simple optimization.

File: prime-router/src/main/kotlin/azure/HttpUtilities.kt [hunk 5]
```diff
@@ -116,6 +142,19 @@ class HttpUtilities {
             return """{"error": "$message"}"""
         }
 
+        private fun addHeaderIfModified(
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/azure/HttpUtilities.kt:145]: Slightly subtle point. The settings table is initially empty and there isn't a last modified time. So, this code only adds a lastModified header if a last modified time exists.

**Review by jimduff-usds — APPROVED** [review]
Thank you!!

**Review by MauriceReeves-usds — APPROVED** [review]

**Pull request merged**